### PR TITLE
[exiv2] Update Exiv2 to 0.28.0

### PR DESCRIPTION
**Pull request by clee-ai**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/exiv2/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -7,6 +7,7 @@ if("@VCPKG_LIBRARY_LINKAGE@" STREQUAL "static")
     if("@EXIV2_ENABLE_NLS@")
         find_package(Intl REQUIRED)
     endif()
+    find_package(unofficial-inih REQUIRED)
```
Comment (JonLiu1993) [inline ports/exiv2/vcpkg-cmake-wrapper.cmake:10]: @CCInc, Thanks for your PR, could you please claify why there is not `find_package(unofficial-inih CONFIG REQUIRED)`?

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/exiv2/vcpkg-cmake-wrapper.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/exiv2/vcpkg-cmake-wrapper.cmake:10]: The `CONFIG` keyword is optional, and the package name is sufficently unique to vcpkg.

(CMake will first look for modules, then for config. If someone wants to play foul on the `unofficial-inih` name, he can do so anyways.)

**Comment by JonLiu1993** [comment]
@CCInc, when I tested the features by command `./vcpkg install exiv2[*]:x86-windows` I get this error.
Please take a look:
````
-- Could NOT find Python3 (missing: Python3_EXECUTABLE Interpreter) 
    Reason given by package: 
        Interpreter: Cannot use the interpreter "C:/Program Files/WindowsApps/PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0/python.exe"

CMake Warning at cmake/findDependencies.cmake:35 (message):
  Python3 was not found.  Python tests under the 'tests' folder will not be
  executed
Call Stack (most recent call fi
````

**Comment by dg0yt** [comment]
> Python3 was not found.  Python tests under the 'tests' folder will not be
>   executed

If possible, disable the tests, don't enable python.

**Comment by CCInc** [comment]
Hi @JonLiu1993 , I made the changes you requested. The python issue is just a warning, they use python as a test suite runner, and we do not run the tests anyways, it should be safe to ignore.

**Comment by JonLiu1993** [comment]
@CCInc, I disabled to find `Python3` but when I re-tested, I encountered a new error, could you please take a look?
https://github.com/Exiv2/exiv2/blob/22b65ccfea31921ef9f96d5ec627433d932364bb/cmake/findDependencies.cmake#L33-L36
````
#find_package (Python3 COMPONENTS Interpreter)
#if (NOT Python3_Interpreter_FOUND)
#    message(WARNING "Python3 was not found. Python tests under the 'tests' folder will not be executed")
#endif()
````

error:
````
CMake Error at xmpsdk/CMakeLists.txt:29 (target_link_libraries):
  Target "exiv2-xmp" links to:

    EXPAT::EXPAT

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at src/CMakeLists.txt:186 (target_link_libraries):
  Target "exiv2lib" links to:

    EXPAT::EXPAT

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done (0.1s)
````

**Comment by JonLiu1993** [comment]
@CCInc, could you please fix the conflicting files?

**Comment by CCInc** [comment]
@JonLiu1993 I fixed building all the features available for Ubuntu, including the expat issue you mentioned.

File: ports/exiv2/fix-expat.patch [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+index 41a672e8..1311c856 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -182,9 +182,7 @@ target_include_directories(exiv2lib SYSTEM PRIVATE
+ )
+ 
+ if (EXIV2_ENABLE_XMP OR EXIV2_ENABLE_EXTERNAL_XMP)
+-    target_include_directories(exiv2lib PRIVATE ${EXPAT_INCLUDE_DIR})
+-    target_link_libraries(exiv2lib PRIVATE EXPAT::EXPAT)
+-    list(APPEND requires_private_list "expat")
```
Comment (dg0yt) [inline ports/exiv2/fix-expat.patch:11]: This line probably needs to stay. I assume it is for the pc file (`Requires.private: expat`).

File: ports/exiv2/fix-inih.patch [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
++++ b/src/CMakeLists.txt
+@@ -262,11 +262,8 @@ if( Iconv_FOUND AND Iconv_LIBRARIES )
+ endif()
+ 
+ if( EXIV2_ENABLE_INIH )
+-  target_link_libraries( exiv2lib_int PRIVATE inih::libinih )
+-  target_link_libraries( exiv2lib_int PRIVATE inih::inireader )
+-  target_link_libraries( exiv2lib PRIVATE inih::libinih )
+-  target_link_libraries( exiv2lib PRIVATE inih::inireader )
+-  list(APPEND requires_private_list "INIReader")
```
Comment (dg0yt) [inline ports/exiv2/fix-inih.patch:40]: This line probably needs to remain, too.
However, vcpkg port `inih` doesn't install a pc file, so the unconditional use of inih breaks consuming the `exiv2` port with `pkg-config`, assuming that it works now.

**Comment by JonLiu1993** [comment]
Pinging @CCInc Can you address the review suggestions? Thanks!

**Comment by JonLiu1993** [comment]
Ping @CCInc for response.

**Comment by JonLiu1993** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**